### ISSUE TITLE
Optimize Nightly sequencing

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#8473]
Every day we have the following sequence:

1. Build GeckoView Nightly
2. Bump GV Nightly version in Android Components
3. Build Android Components Nightly
4. Bump Android Components version in Fenix
5. Build Fenix Nightly.

Right now the time they are build is not optimal, meaning that a GeckoView Nightly build today will be in tomorrow's Fenix Nightly at the earliest. Now, that 1-3 are already fully automated, with Mergify merging the PRs, we should look into optimizing the sequence. So that without build/test errors we get all of that out the same day.

**Comment by pocmo** [comment]
Picking some random GV Nightly versions on Maven I see the following timestamps: 12:17:42, 12:44:29, 12:07:08, 12:58:06, 11:53:46, 10:10:47, 12:40:15

**Comment by pocmo** [comment]
I guess we have two options:
- (1) Pick a time for the hooks so that they will be likely to run after the previous one has completed.
- (2) Trigger the next hook from the previous hook?

I guess (1) is easy, although maybe not perfect, and (2) is harder to get setup.

@JohanLorenzo what do you think?

**Comment by JohanLorenzo** [comment]
I agree, (1) is definitely doable. We can change the timing today and see how it goes.

(2) looks feasible. We would have to spin a new task in each nightly graph. That task would talk to Taskcluster and trigger the right hook. Triggering hooks is the cornerstone of release promotion. So I don't see any issues with that. What do you think @escapewindow?

**Comment by pocmo** [comment]
I think one additional issue with  option (2) is that the steps (2) and (4) mentioned above currently open a PR that needs to be merged before we can continue and the chain of tasks would need to wait for that to be merged. The merge (2) is already happening automatically [I want to do that with (4) too soon], but still it will take a bit until that has happened.

**Comment by escapewindow** [comment]
> (2) looks feasible. We would have to spin a new task in each nightly graph. That task would talk to Taskcluster and trigger the right hook. Triggering hooks is the cornerstone of release promotion. So I don't see any issues with that. What do you think @escapewindow?

I think so. I'm leaning slightly more towards pulse-triggering the hook on upstream graph success, but we can investigate hook-triggering tasks as well.

**Comment by JohanLorenzo** [comment]
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1667367 to handle option (1).

**Comment by JohanLorenzo** [comment]
@pocmo, @rvandermeulen, @jcristau, and I agreed to try this schedule for solution (1) - times are UTC:

1. [mozilla-central] ship GV starts at 10am (same as it is now)
2. [android-components] GV bump at 2pm
3. [android-components] AC nightly at 2:30pm
4. [fenix] AC bump at 3:30pm
5. [fenix] Fenix at 5pm
